cmd/vulcan-host-discovery/main.go

```diff
@@ -132,23 +132,24 @@ nmapReportLoop:
 			hostHostnames = append(hostHostnames, hostname.Name)
 		}
 
-		// At this point the host is not in the list of known hosts, so add the host to the results.
-		add = true
+		vuln := exposedVuln
 
 		networkResource := map[string]string{
 			"Subnet":     targetSubnet,
 			"IP Address": strings.Join(hostIPAddresses, ","),
 			"Hostname":   strings.Join(hostHostnames, ","),
 		}
-		gr.Rows = append(gr.Rows, networkResource)
-	}
+		gr.Rows = []map[string]string{networkResource}
+
+		vuln.AffectedResource = networkResource["IP Address"]
+		vuln.Resources = []report.ResourcesGroup{gr}
```

Comment:
I think we still have the problem here, because we are reusing the same `ResourcesGroup` in all the rounds of the loop. We should move the creation of the `gr` inside the loop (from L77 to L137, for example).

Reply:
My bad, in this case there's no problem by doing it this way, sorry :)